frontend/apps/web/components/RunTImeline/RunTimeline.tsx

```diff
@@ -151,139 +147,249 @@ export default function RunTimeline(props: Props): ReactElement {
+                    task={task}
+                    index={index}
+                    jobStatus={jobStatus}
+                    timelineStart={timelineStart}
+                    totalDuration={totalDuration}
+                    topOffset={topOffset}
+                    expandedTaskId={expandedTaskId}
+                    toggleExpandedRowBody={toggleExpandedRowBody}
+                    isExpanded={isExpanded}
+                    isLastItem={isLastItem}
+                  />
+                </React.Fragment>
               );
             })}
           </div>
         </div>
       </div>
     </div>
   );
+}
 
-  interface TableHeaderProps {
-    formatDate: (date: Date) => string;
-    getPositionPercentage: (
-      time: Date,
-      timelineStart: Date,
-      totalDuration: number
-    ) => number;
-    timeLabels: Date[];
-    timelineStart: Date;
-    totalDuration: number;
-  }
+interface LeftActivityBarProps {
+  filteredTasks: JobRunEvent[];
+  toggleExpandedRowBody: (val: string) => void;
+  jobStatus: JobRunStatus | undefined;
+  expandedTaskId: string;
+}
+
+function LeftActivityBar(props: LeftActivityBarProps): ReactElement {
+  const { filteredTasks, toggleExpandedRowBody, jobStatus, expandedTaskId } =
+    props;
+  return (
+    <div>
+      <div className="sticky top-0 h-14 bg-gray-200 dark:bg-gray-800 z-10 px-6 border-b border-gray-200 dark:border-gray-700" />
+      <div className="border-r border-gray-200 dark:border-gray-700 flex flex-col text-sm">
+        {filteredTasks.map((task, index) => {
+          const isLastItem = index === filteredTasks.length - 1;
+          const isExpanded = expandedTaskId === String(task.id);
+          return (
+            <div
+              className={cn(
+                'px-2 h-10 items-center flex cursor-pointer hover:bg-gray-100 dark:hover:bg-gray-700',
+                !isLastItem && 'border-b border-gray-200 dark:border-gray-700',
+                isExpanded && 'h-[165px]'
+              )}
+              onClick={() => toggleExpandedRowBody(String(task.id))}
+              key={task.id}
+            >
+              <ActivityLabel
+                task={task}
+                getStatus={() => getTaskStatus(task, jobStatus)}
+              />
+            </div>
+          );
+        })}
+      </div>
+    </div>
+  );
+}
+
+interface TimelineBarProps {
+  index: number;
+  timelineStart: Date;
+  totalDuration: number;
+  task: JobRunEvent;
+  jobStatus: JobRunStatus | undefined;
+  topOffset: number;
+  expandedTaskId: string | null;
+  toggleExpandedRowBody: (val: string) => void;
+  isExpanded: boolean;
+  isLastItem: boolean;
+}
+
+function TimelineBar(props: TimelineBarProps) {
+  const {
+    task,
+    jobStatus,
+    timelineStart,
+    totalDuration,
+    topOffset,
+    toggleExpandedRowBody,
+    isExpanded,
+    isLastItem,
+  } = props;
+
+  const scheduled = task.tasks.find(
+    (st) => st.type == 'ActivityTaskScheduled'
+  )?.eventTime;
```

Comment:
same question again here. why are we always looking at this as opposed to just looking at the tasks start time?

Reply:
explanation above ^^